app/ui/src/app/integration/edit-page/current-flow.service.ts

```diff
@@ -578,6 +587,40 @@ export class CurrentFlowService {
     }
   }
 
+  executeStepCustomizations(
+    position: number,
+    step: Step,
+    then: (step: Step) => void
+  ): any {
+    switch (step.stepKind) {
+      case SPLIT:
+      case AGGREGATE:
+        // A split step needs the data shape of the previous thing with a data shape
+        const prev = this.getPreviousStepWithDataShape(position);
+        this.integrationSupportService
+          .getStepDescriptor(SPLIT, {
```

Comment:
@gashcrumb I think `step.stepKind` should be passed to `getStepDescriptor` as a first argument

Reply:
see PR for this: https://github.com/syndesisio/syndesis/pull/4561